### PR TITLE
Use new kompile --haskell-backend-command when building tests

### LINKED ISSUE #3100: Fix Makefile test infrastructure
Our CI integration tests should use our kore-exec build when running kompile instead of the one shipped with the frontend.

This should use the new K dependency: https://github.com/runtimeverification/haskell-backend/pull/3043. That PR also requires this change to the test infrastructure for it to pass. The new frontend adds the `--haskell-backend-command` option to kompile which we should use with our version of kore-exec.

### PR DESCRIPTION
**Pull request by JKTKops**
Fixes #3100

### Scope:

Modifies the makefile and regression test builder script to use our version of `kore-exec` instead of the version shipped with `kompile`.

### Estimate:

6/22

---

###### Review checklist

The author performs the actions on the checklist. The reviewer evaluates the work and checks the boxes as they are completed.

-   [x] **Summary.** Write a summary of the changes. Explain what you did to fix the issue, and why you did it. Present the changes in a logical order. Instead of writing a summary in the pull request, you may push a clean Git history.
-   [x] **Documentation.** Write documentation for new functions. Update documentation for functions that changed, or complete documentation where it is missing.
-   [x] **Tests.** Write unit tests for every change. Write the unit tests that were missing before the changes. Include any examples from the reported issue as integration tests.
-   [x] **Clean up.** The changes are already clean. Clean up anything near the changes that you noticed while working. This does not mean only spatially near the changes, but logically near: any code that interacts with the changes!

### CODE REVIEW

**Comment by JKTKops** [comment]
This PR now has CI correctly using the `--backend-haskell-command` option, however, attempting to build the tests results in this error:
```
$ make test
>>> /home/max/Documents/k/haskell-backend/test/issue-3074 kompile test.k
rm -fr ./test-kompiled
kompile --backend haskell --haskell-backend-command /home/max/Documents/k/haskell-backend/.build/kore/bin/kore-exec -d . test.k


kore-exec definition.kore --module TEST --output haskellDefinition.bin --serialize
KORE_EXEC_OPTS='--no-bug-report --disable-stuck-check'

Missing: --prove SPEC_FILE --spec-module SPEC_MODULE

Usage: kore-exec [--version] 
                 { ... lots of options ... }
                   [--[no-]serialize] [--simplifierx]]
  Uses Kore definition in DEFINITION_FILE to execute pattern in PATTERN_FILE.
[Error] Critical: Haskell backend reported errors validating compiled
definition.
Examine output to see errors.
[Warning] Compiler: Could not find main syntax module with name TEST-SYNTAX in
definition.  Use --syntax-module to specify one. Using TEST as default.
make: *** [/home/max/Documents/k/haskell-backend/test/issue-3074/../include.mk:50: test-kompiled/definition.kore] Error 113
```

This looks to me like an issue with how `kompile` calls `kore-exec` (since the makefile does not invoke `kore-exec` there). It's also possible that it's an issue with `--serialize`, but I don't think so, since `--serialize` works fine when not using kframework v5.3.80.

**Comment by JKTKops** [comment]
Dwight noted that `KORE_EXEC_OPTS` here contains a proof option, even though `--prove` is not present. `optparse-applicative` does not like that, and generates an error like this one. I'm not sure if this is the error I'd expect, but I do think that erroring is probably the correct behavior. I've modified one of the `include.mk` makefiles further to unset that environment variable while calling `kompile`.

File: include.mk [hunk 1]
```diff
@@ -19,7 +19,7 @@ export KRUN
 KPROVE = kprovex
```
[the remark this answers is not in the document]
Comment (ana-pantilie) [reply include.mk:19]: We actually need this in order to remove kprovex, so this will come in a separate PR.

**Review by ana-pantilie — APPROVED** [review]
🎉

**Pull request merged**